src/main/java/com/sourcery/gymapp/backend/userProfile/repository/UserProfileRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.sourcery.gymapp.backend.userProfile.repository;
+
+
+import com.sourcery.gymapp.backend.userProfile.model.UserProfile;
+import java.util.UUID;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface UserProfileRepository extends JpaRepository<UserProfile, UUID> {
+
+    UserProfile findUserProfileByUserId(UUID userId);
```

Comment:
It is a better practice in repository to return an Optional<UserProfile>.
 Then, instead of checking for null manually in the service, we can use orElseThrow for better null handling:
 UserProfile userProfile = userProfileRepository
        .findUserProfileByUserId(currentUserId)
        .orElseThrow(() -> new UserProfileNotFoundException(currentUserId));